javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -118,18 +41,157 @@ object Pattern {
       res
     }
 
-  def adaptPatternToRe2(regex: String): String = {
-    regex
+  private object CompiledPatternStore {
+    private case class Entry(regex: String, flags: Int, re2: RE2RegExpOps)
```

Comment:
We don't expose our RE2 as a public API, so flags are always constant, no need to remember them in the cache. It can be as simple as a mapping from regex string to a compiled regex instance.

Reply:
The overload `compile(String regex, int flags)` does exist in JDK. `Entry.flags` is meant to cache the second parameter (not RE2's).

Reply:
Good point.